packages/@react-aria/interactions/src/utils.ts

```diff
@@ -12,57 +12,25 @@
 
 import {FocusableElement} from '@react-types/shared';
 import {focusWithoutScrolling, getOwnerWindow, isFocusable, useEffectEvent, useLayoutEffect} from '@react-aria/utils';
-import {FocusEvent as ReactFocusEvent, useCallback, useRef} from 'react';
-
-export class SyntheticFocusEvent<Target = Element> implements ReactFocusEvent<Target> {
-  nativeEvent: FocusEvent;
-  target: EventTarget & Target;
-  currentTarget: EventTarget & Target;
-  relatedTarget: Element;
-  bubbles: boolean;
-  cancelable: boolean;
-  defaultPrevented: boolean;
-  eventPhase: number;
-  isTrusted: boolean;
-  timeStamp: number;
-  type: string;
-
-  constructor(type: string, nativeEvent: FocusEvent) {
-    this.nativeEvent = nativeEvent;
-    this.target = nativeEvent.target as EventTarget & Target;
-    this.currentTarget = nativeEvent.currentTarget as EventTarget & Target;
-    this.relatedTarget = nativeEvent.relatedTarget as Element;
-    this.bubbles = nativeEvent.bubbles;
-    this.cancelable = nativeEvent.cancelable;
-    this.defaultPrevented = nativeEvent.defaultPrevented;
-    this.eventPhase = nativeEvent.eventPhase;
-    this.isTrusted = nativeEvent.isTrusted;
-    this.timeStamp = nativeEvent.timeStamp;
-    this.type = type;
-  }
-
-  isDefaultPrevented(): boolean {
-    return this.nativeEvent.defaultPrevented;
-  }
-
-  preventDefault(): void {
-    this.defaultPrevented = true;
-    this.nativeEvent.preventDefault();
-  }
-
-  stopPropagation(): void {
-    this.nativeEvent.stopPropagation();
-    this.isPropagationStopped = () => true;
-  }
-
-  isPropagationStopped(): boolean {
-    return false;
-  }
+import {FocusEvent as ReactFocusEvent, SyntheticEvent, useCallback, useRef} from 'react';
+
+// Turn a native event into a React synthetic event.
+export function createSyntheticEvent<E extends SyntheticEvent>(nativeEvent: Event): E {
+  let event = nativeEvent as any as E;
+  event.nativeEvent = nativeEvent;
+  event.isDefaultPrevented = () => event.defaultPrevented;
+  // cancelBubble is technically deprecated in the spec, but still supported in all browsers.
+  event.isPropagationStopped = () => (event as any).cancelBubble;
+  event.persist = () => {};
+  return event;
+}
 
-  persist() {}
+export function setEventTarget(event: Event, target: Element) {
```

Comment:
We probably need to add composedPath somehow... and the target can't necessarily be target, it might need to be the shadow root, though depends on which side of the shadow root we are? it's hurting my head a bit, lets discuss this week